source/slang/lexer.cpp

```diff
@@ -84,22 +84,22 @@ namespace Slang
     // Lexer
 
     void Lexer::initialize(
-        SourceFile*     inSourceFile,
+        SourceUnit*     inSourceUnit,
         DiagnosticSink* inSink,
         NamePool*       inNamePool)
     {
-        sourceFile  = inSourceFile;
+        sourceUnit  = inSourceUnit;
         sink        = inSink;
         namePool    = inNamePool;
 
-        auto content = inSourceFile->content;
+        auto content = inSourceUnit->getSourceFile()->content;
```

Comment:
Small detail: it might be nice for `SourceUnit` to provide a `getContent` that forwards to `SourceFile` so that people don't need to chain like this.